packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -345,6 +351,14 @@ export async function activate(context: vscode.ExtensionContext) {
     sfdxProjectOpened = files && files.length > 0;
   }
 
+  let isChangeSetWorkspace = false;
```

Comment:
Adding an additional check that is used to restrict exposing the source retrieve and deploy commands to change-set-based projects only.